File: JS-Solutions/Problem-07.js
// What is the 10 001st prime number?

const isPrime = (num) => {
  if (num < 2) {
    return false;
  }
  if (num == 2) {
    return true;
  }
  for(let i=2;i<=Math.sqrt(num)+1;i++){
    if(num % i == 0){
      return(false);
    }
  }
  return(true);
}

const nthPrime = (limit) => {
  let prime_list = [];
  let num = 2;
  while (prime_list.length < limit) {
    if (isPrime(num)) {
      prime_list.push(num);
    }
    num += 1;
  }
  return prime_list[prime_list.length - 1];
}

console.log(nthPrime(10001))

File: Py-Solutions/Problem-07.py
# What is the 10,001st prime number?
import math

def isPrime(num):
  if num < 2:
    return False
  elif num ==2:
    return True
  else:
    return sum([i for i in range(1, int(num**0.5) + 1) if num % i == 0]) == 1

def primeFinder(n):
  counter = 1
  nth = 1
  while nth < n:
    counter += 2
    if isPrime(counter):
      nth += 1
  nth_prime = counter
  return nth_prime

def main():
  print(primeFinder(10001))

if __name__ == '__main__':
  main()
